vendor/processors/stripe_processor.py

```diff
@@ -352,12 +364,15 @@ def build_price(self, offer, price):
             price_data['recurring'] = {
                 'interval': 'month' if offer.term_details['term_units'] == TermDetailUnits.MONTH else 'year',
                 'interval_count': offer.term_details['payment_occurrences'],
-                'usage_type': 'license'
+                'usage_type': 'licensed'
             }
         
         return price_data
     
     def build_coupon(self, offer, price):
+        if isinstance(price, (int, float)):
+            raise TypeError(f"Price cannot be created without a valid vendor price object added to this offer")
+
         coupon_data = {
             'name': offer.name,
             'currency': price.currency,
```

Comment:
If you do not have a price instance because the offer does not have one you can use this value for the currency


`from vendor.config import DEFAULT_CURRENCY`